packages/npm-dependencies-validation/test/utils/npmUtil.spec.ts

```diff
@@ -1,15 +1,55 @@
+import { fail } from "assert";
 import { expect } from "chai";
+import { createSandbox, SinonSpy } from "sinon";
+import { NpmLsResult, VscodeOutputChannel } from "../../src/types";
 import { invokeNPMCommand, getNPM } from "../../src/utils/npmUtil";
 
 describe("npmUtil unit test", () => {
-  it("invokeNPMCommand failed", () => {
-    return expect(
-      invokeNPMCommand(["ls"], "./non_existing_path")
-    ).to.be.rejectedWith(`spawn ${getNPM()} ENOENT`);
+  const sandbox = createSandbox();
+  const outputChannel: VscodeOutputChannel = {
+    append: (data: string) => console.log(data),
+  };
+
+  afterEach(function () {
+    sandbox.restore();
   });
 
-  it("invokeNPMCommand succeeded with ls", () => {
-    return expect(invokeNPMCommand(["ls", "--depth=0"], "./")).to.be.fulfilled;
+  describe("invokeNPMCommand failed", () => {
+    it("outputChannel is not provided", () => {
+      const config = { commandArgs: ["ls"], cwd: "./non_existing_path" };
+      return expect(invokeNPMCommand(config)).to.be.rejectedWith(
+        `spawn ${getNPM()} ENOENT`
+      );
+    });
+
+    it("outputChannel is provided", () => {
+      const appendSpy: SinonSpy = sandbox.spy(outputChannel, "append");
+      const config = { commandArgs: ["ls"], cwd: "./non_existing_path" };
+      return invokeNPMCommand(config, outputChannel)
```

Comment:
it would be more **declarative and less error prone** to use `chai-as-promised` and `to.be.rejectedWith(...)`
instead of `.then / .catch`
Not sure how to combine it with the assertion on the sinonSpy, but may be worth a shot.

Reply:
What happens when you do?

```typescript
async () => {
      const appendSpy: SinonSpy = sandbox.spy(outputChannel, "append");
      const config = { commandArgs: ["ls"], cwd: "./non_existing_path" };
      await invokeNPMCommand(config, outputChannel).to.be.rejectedWith(
        `spawn ${getNPM()} ENOENT`
      );
      expect(appendSpy.called).to.be.true; 
```

Reply:
It will not work. According to chai-as-promised API I need to use return before invokeNPMCommand